docs/source/security/cors.mdx

```diff
@@ -0,0 +1,123 @@
+
+It's common knowledge that all Internet users need to think carefully before installing any new software on their devices. However, when it comes to websites, we navigate to different sites all the time, letting our browsers load content from those sites along the way.
+
+As web developers, we aim to have users navigate to our website, but we don't want the user's browser to do anything fishy to our server while the user is visiting. Browser security mechanisms (e.g., CORS or SOP) can give developers peace of mind by enabling a website's server to specify which browser's origins can request resources from that server.
+
+The [_origin_](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy#definition_of_an_origin) of a piece of web content consists of that content's domain, protocol, and port. The [same-origin policy](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) (SOP) is a security mechanism that restricts scripts on one origin from interacting with resources from another origin.
+
+
+```bash
+# Same origin
+http://example.com:8080/ <==> http://example.com:8080/
+
+# Different origin (difference in domain, protocol, and port)
+http://example.com:8080/ =X= https://example1.com:8081/
+```
+
+However, as we all know, the internet is an exciting place full of resources that can make websites better (importing images, extra fonts, making API calls, etc.).  Developers needed a new protocol to relax SOP and safely _share_ resources across different origins.
+
+Cross-Origin Resource Sharing is the mechanism that allows a web page to  _share_ resources across different origins. CORS provides an extra layer of protection by enabling servers and clients to define HTTP headers that specify _which_ external clients' scripts can access their resources.
+
+Note that both SOP and CORS are related to [website and browser security](https://developer.mozilla.org/en-US/docs/Web/Security#security-related_glossary_terms); neither prevents other kinds of software from requesting resources from your server.
+
+## Configuring CORS options for Apollo Server
+
+By default, the batteries-included `apollo-server` package uses the wildcard value (`*`) to allow requests from any origin _without credentials_ to access its served responses.  We'll go into more details about what this means in a moment.
+
+The batteries-included `apollo-server` [has a `cors` option](../api/apollo-server/#cors), which you can use to change your server's default CORS behavior:
+
+```js
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  cors: {
+    origin: ["http://localhost:3000", "https://studio.apollographql.com"]
+  },
+});
+```
+
+The example above enables CORS requests from two origins `http://localhost:3000` and `https://studio.apollographql.com` (the origin for [Apollo Studio Explorer](https://www.apollographql.com/docs/studio/explorer/explorer/)).
```

Comment:
`http://localhost:3000` and `https://studio.apollographql.com` -> `http://localhost:3000`, along with `https://studio.apollographql.com`

(This change makes the parenthetical more cleanly apply only to the second URL)

Reply:
Ah and cut "two origins"